src/snowcli/app/dev/commands_structure.py

```diff
@@ -11,12 +12,27 @@ class _Node:
     name: str
     children: Dict[str, _Node] = field(default_factory=dict)
     level: int = 0
+    options: Dict[str, List[str]] = field(default_factory=dict)
 
     def print(self):
         print("    " * self.level, self.name)
         for ch in self.children.values():
             ch.print()
 
+    def print_with_options(self):
+        options = self._prepare_options_dict()
+        pretty = json.dumps(options, indent=4)
+        print(pretty)
```

Comment:
Can you show example result of this callback?

Reply:
It`s pretty huge, but for one command and global options, it looks like:
`{
    "options": {
        "version": [
            "--version"
        ],
        "docs": [
            "--docs"
        ],
        "structure": [
            "--structure"
        ],
        "options_structure": [
            "--options-structure"
        ],
        "info": [
            "--info"
        ],
        "configuration_file": [
            "--config-file"
        ],
        "pycharm_debug_library_path": [
            "--pycharm-debug-library-path"
        ],
        "pycharm_debug_server_host": [
            "--pycharm-debug-server-host"
        ],
        "pycharm_debug_server_port": [
            "--pycharm-debug-server-port"
        ],
        "disable_external_command_plugins": [
            "--disable-external-command-plugins"
        ],
        "commands_registration": [
            "--commands-registration"
        ],
        "install_completion": [
            "--install-completion"
        ],
        "show_completion": [
            "--show-completion"
        ]
    },
    "connection": {
        "options": {
            "install_completion": [
                "--install-completion"
            ],
            "show_completion": [
                "--show-completion"
            ]
        },
        "list": {
            "options": {
                "format": [
                    "--format"
                ],
                "verbose": [
                    "--verbose",
                    "-v"
                ],
                "debug": [
                    "--debug"
                ]
            }
        },
        "add": {
            "options": {
                "connection_name": [
                    "--connection-name",
                    "-n"
                ],
                "account": [
                    "--account",
                    "-a",
                    "--accountname"
                ],
                "user": [
                    "--user",
                    "-u",
                    "--username"
                ],
                "password": [
                    "--password",
                    "-p"
                ],
                "role": [
                    "--role",
                    "-r"
                ],
                "warehouse": [
                    "--warehouse",
                    "-w"
                ],
                "database": [
                    "--database",
                    "-d"
                ],
                "schema": [
                    "--schema",
                    "-s"
                ],
                "host": [
                    "--host",
                    "-h"
                ],
                "port": [
                    "--port",
                    "-P"
                ],
                "region": [
                    "--region",
                    "-R"
                ],
                "format": [
                    "--format"
                ],
                "verbose": [
                    "--verbose",
                    "-v"
                ],
                "debug": [
                    "--debug"
                ]
            }
        },
        "test": {
            "options": {
                "connection": [
                    "--connection",
                    "-c",
                    "--environment"
                ],
                "format": [
                    "--format"
                ],
                "verbose": [
                    "--verbose",
                    "-v"
                ],
                "debug": [
                    "--debug"
                ]
            }
        }
    }
}   `

Reply:
Do we need it as command callbck? The other callback we have have some use-cases (like copying to doc), But I'm not sure about this one

Reply:
It seemed reasonable, to place it there - with command structure, version etc.
But i`m open to suggestions :)

Reply:
Do we plan to use it other way than in tests?

Maybe we can call `generate_commands_structure` directly from tests?

Reply:
WDYT?